engine/src/main/java/org/opencds/cqf/cql/engine/elm/execution/CollapseEvaluator.java

```diff
@@ -104,8 +104,15 @@ public static List<Interval> collapse(Iterable<Interval> list, Quantity per, Con
         {
             Interval applyPer = getIntervalWithPerApplied(intervals.get(i), per, context);
 
-            if(isTemporal) {
-                applyPer = intervals.get(i);
+            if (isTemporal) {
+                if (per.getValue().compareTo(BigDecimal.ONE) == 0 || per.getValue().compareTo(BigDecimal.ZERO) == 0) {
```

Comment:
See the discussion here, but this is likely incorrect:
https://github.com/DBCG/cql_engine/pull/499

Reply:
I'd also prefer to bug fixes in separate PRs if possible.